src/Unidevel.OpenWeather/OpenWeatherClient.cs

```diff
@@ -9,34 +9,46 @@ public class OpenWeatherClient : IDisposable, IOpenWeatherClient
     {
         private readonly IConfiguration _configuration;
         private readonly string _apiKey;
+        private bool _disposed = false;
+
+        private readonly HttpClient _httpClient = new HttpClient();
 
         public OpenWeatherClient(IConfiguration configuration = null, string apiKey = null)
         {
             _configuration = configuration;
-            _apiKey = apiKey ?? configuration?.GetSection("OpenWeather")?["ApiKey"];            
+            _apiKey = apiKey ?? configuration?.GetSection("OpenWeather")?["ApiKey"];
         }
 
         public async Task<CurrentWeather> GetCurrentWeatherAsync(float longitude = float.NaN, float latitude = float.NaN, string cityNameCountryCode = null, int cityId = Int32.MinValue, string apiKey = null)
         {
             var url = buildCurrentWeatherUrl(apiKey ?? _apiKey, longitude, latitude, cityNameCountryCode, cityId);
 
-            string currentWeatherJson;
-            using (var webClient = new WebClient()) currentWeatherJson = await webClient.DownloadStringTaskAsync(url);
-
-            var currentWeather = Newtonsoft.Json.JsonConvert.DeserializeObject<CurrentWeather>(currentWeatherJson);
-
-            return currentWeather;
+            try
+            {
+                var response = await _httpClient.GetStringAsync(url);
+                var currentWeather = Newtonsoft.Json.JsonConvert.DeserializeObject<CurrentWeather>(response);
+                return currentWeather;
+            }
+            catch (HttpRequestException)
+            {
+                throw;
```

Comment:
It does nothing. Do not use try/catch clause here. Errors are handled well without it. Sometimes best error handling is not to implement anything. Exceptions should be catched only where you know what you want to do with them.